client_test.go

```diff
@@ -222,6 +224,28 @@ func TestS3PointerUnmarshalInvalidLength(t *testing.T) {
 	assert.ErrorContains(t, err, "invalid pointer format, expected length 2, but received [3]")
 }
 
+func TestWithObjectPrefix(t *testing.T) {
```

Comment:
I noticed that for this tests, you don't actually need to call the `New` constructor to cause your function to return an error, right?  So, you can probably just concentrate on your `WithPrefix` function and make simpler.

Reply:
Exactly. 
The only "problem" of using directly `WithObjectPrefix`, is that we can't test that `c.objectPrefix` contains the value.
But probably is not bad, If we don't return an error, then that means that we store the prefix in `c.objectPrefix`.
What do u think? I think that is fine if we use the new function directly and make this test simpler

Reply:
That sounds perfect. I'll split it